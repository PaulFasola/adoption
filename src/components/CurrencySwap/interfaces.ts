import { IProtocol } from '../ProtocolSelector/interfaces';

export interface IProtocolArrayPipe {
  input: IProtocol[];
  output: IProtocol[];
}

export interface IProtocolPipe {
  input?: IProtocol;
  output?: IProtocol;
}

export interface IProps {
  locked?: boolean;
  noShadow?: boolean;
  protocols: Partial<IProtocolArrayPipe>;
  maxFractionDigits?: number;

  onSubmit?: () => void;
}

export interface ISwapValues {
  input?: number;
  output?: number;
}
