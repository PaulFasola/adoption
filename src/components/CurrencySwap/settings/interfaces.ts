import { IProps as IInputProps } from '../../common/Input/interfaces';
import { IProps } from '../interfaces';

export interface ISettingsProps extends Pick<IProps, 'settings'> {
  settings: ISettings;
  visible?: boolean;
  renderable?: boolean;
}

export interface ISettings {
  [key: string]: ISetting;
}

export type SettingType = 'boolean' | 'string' | 'number' | 'radio';

export interface ISetting {
  visible: boolean;
  type: SettingType;
  label: string;
  hint?: string;
  customInput?: IInputProps;
  value?: any;
}
