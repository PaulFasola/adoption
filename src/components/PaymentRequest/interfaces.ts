import { PaymentStatus } from "./enums/paymentStatus";

export interface IProps {
	address: string;
	symbol: string;
	amount: {
		toPay: number;
		received?: number;
	}
	logos: {
		coin: string;
		company?: string;
	};
	decimalPlaces: number;
	waitAnimation?: boolean;
	sellerName?: string;
	strings?: Partial<IStrings>;
	onCancel?: () => void;
	helpUrl?: string;
	showQRCode?: boolean;
	transactions?: ITransaction[];
	customStatusText?: string;
	status?: PaymentStatus;
	deadline?: {
		dateLocale?: string;
		datetime: Date;
		humanized?: boolean;
	};
}

export interface IStrings {
	request: string;
	cancel: string;
	seller: string;
	status: string;
	help: string;
	txStatus: {
		pending: string;
		complete: string;
		failed: string;
	}
	deadline: string;
	receivedAmount: string;
	remainingAmount: string;
	transactions: string;
}

export interface ITransaction {
	txHash: string;
	txUrl: string;
	amount: number;
}