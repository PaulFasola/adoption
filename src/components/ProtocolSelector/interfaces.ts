export interface IProtocol {
  logoURI: string;
  name: string;
  decimals: number;
  symbol: string;
  value: number;
  balance?: number;
}

export interface IProps {
  disabled?: boolean;
  current?: IProtocol;
  list: IProtocol[];
  allowCustom?: boolean;

  onChange?: (newProtocol: IProtocol | null) => void;
}
