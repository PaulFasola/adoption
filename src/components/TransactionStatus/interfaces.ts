import { TxStatus } from "./txStatus";

export interface IProps {
	amount: string;
	symbol: string;
	status?: TxStatus;
	animated?: boolean;
	date?: {
		value?: Date,
		locale?: string,
		options?: Intl.DateTimeFormatOptions
	};
	txURL?: string;
	colorMap?: Record<TxStatus, string>;
	uncapitalizeStatus?: boolean;
	autoShowDetails?: boolean;

	// TxDetails
	sender?: IAdress;
	receiver?: IAdress;
	txFees?: string;
	customDetailComponent?: React.ReactNode;
}

export interface IAdress {
	hash: string;
	url?: string;
}