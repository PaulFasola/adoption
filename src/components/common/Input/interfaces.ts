export interface InputProps {
  id?: string;
  pattern?: string;
  className?: string;
  maxLength?: number;
  minLength?: number;
  max?: number;
  min?: number;
  placeholder?: string;
  title?: string;
  value?: string | number;
  label?: string;
  autoComplete?: 'on' | 'off';
  autoFill?: 'on' | 'off';
  style?: React.CSSProperties;

  onValueChange?: (value: string) => void;
}

export interface IProps extends InputProps {
  type: 'text' | 'number' | 'decimal';
  maximumFractionDigits?: number;
  style?: React.CSSProperties;
}
