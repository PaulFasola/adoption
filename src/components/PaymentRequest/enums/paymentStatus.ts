export enum PaymentStatus {
	COMPLETE = 'complete',
	PENDING = 'pending',
	FAILED = 'failed'
}