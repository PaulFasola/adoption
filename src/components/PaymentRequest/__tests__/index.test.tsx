import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import { render } from '@testing-library/react';
import { PaymentRequest } from '../';
import { ITransaction } from '../interfaces';
import { configure, shallow } from 'enzyme';
import { PaymentStatus } from '../enums/paymentStatus';
import { advanceBy, advanceTo, clear } from 'jest-date-mock';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const companyAsset = require('../../assets/fictiveCompany.png');

configure({ adapter: new Adapter() });

const testRelativeDateTime = (unit: string, offset: number): void => {
  it(`should render relative time correctly (${unit})`, () => {
    // Let's mock that date
    const mockedDate = new Date(2009, 1, 3, 0, 0, 0);
    advanceTo(mockedDate);

    // Adds 2 hours to the current timestamp
    advanceBy(offset);
    const future = new Date();

    // Go back into the past
    advanceTo(mockedDate);

    /**
     * date   = 2009, 1, 3, 0, 0, 0
     * future = 2009, 1, 3, 2, 0, 0
     */
    const wrapper = shallow(
      <PaymentRequest
        deadline={{ datetime: future, humanized: true }}
        symbol='BTC'
        decimalPlaces={8}
        logos={{
          coin: "https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
        }}
        address="1BitcoinEaterAddressDontSendf59kuE"
        amount={{ toPay: 0.9 }}
      />);

    expect(toJson(wrapper)).toMatchSnapshot();
    clear();
  });
}

describe(`PaymentRequest Basic`, () => {

  it('should render', () => {
    shallow(
      <PaymentRequest
        symbol='BTC'
        decimalPlaces={8}
        logos={{
          coin: "https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
        }}
        address="1BitcoinEaterAddressDontSendf59kuE"
        amount={{ toPay: 0.9 }}
      />);
  });

  // "Should render relative time correctly {unit}"
  testRelativeDateTime('minutes', 2 * 60 * 1000);
  testRelativeDateTime('hours', 2 * 60 * 60 * 1000);
  testRelativeDateTime('days', 2 * 24 * 60 * 60 * 1000);

  it('should render a visual animation if status is "completed" or "failed"', () => {
    [PaymentStatus.COMPLETE, PaymentStatus.FAILED].forEach(status => {
      const wrapper = shallow(
        <PaymentRequest
          status={status}
          symbol='BTC'
          decimalPlaces={8}
          logos={{
            coin: "https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
          }}
          address="1BitcoinEaterAddressDontSendf59kuE"
          amount={{ toPay: 0.9 }}
        />);

      expect(toJson(wrapper)).toMatchSnapshot();
    })
  });

  it('should not render any visual status if 1) payment status is not "completed" or "failed" and 2) showQRCode prop is false.', () => {
    const wrapper = shallow(
      <PaymentRequest
        showQRCode={false}
        symbol='BTC'
        decimalPlaces={8}
        logos={{
          coin: "https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
        }}
        address="1BitcoinEaterAddressDontSendf59kuE"
        amount={{ toPay: 0.9 }}
      />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('PaymentRequest Detailed', () => {
  it('should render', () => {
    shallow(
      <PaymentRequest
        symbol='BTC'
        decimalPlaces={8}
        sellerName='Such Company LTD'
        logos={{
          coin: "https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg",
          company: companyAsset
        }}
        helpUrl="http://foobar/"
        address="1BitcoinEaterAddressDontSendf59kuE"
        amount={{ toPay: 0.9, received: 0 }}
        transactions={[
          {
            txHash: '369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1',
            txUrl: 'https://www.blockchain.com/btc/tx/369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1',
            amount: 0.1
          },
          {
            txHash: '123',
            txUrl: 'https://www.blockchain.com/btc/tx/519f6c9581ce27e0a59f5f8e427b672087e1f2eb1aead0d66288de62ed3e9647',
            amount: 0.1
          }
        ]}
        deadline={{
          dateLocale: 'en-US',
          datetime: new Date(),
          humanized: true
        }}
        strings={{
          help: 'FooBar'
        }}
      />);
  });

  it('should be fully string customizable', () => {
    const wrapper = shallow(
      <PaymentRequest
        symbol='BTC'
        decimalPlaces={8}
        logos={{
          coin: "https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
        }}
        address="1BitcoinEaterAddressDontSendf59kuE"
        amount={{ toPay: 0.9 }}
        strings={{
          cancel: 'Very cancelling',
          seller: 'Seller',
          deadline: 'Deadline',
          // txStatus is overriden by customStatusText props in this example.
          txStatus: {
            complete: 'Done \o/',
            failed: 'Failed :(',
            pending: 'is pending...'
          },
          request: 'Pay {amount} {symbol} to get your item',
          help: 'Wow, help here',
          status: 'Transaction',
          receivedAmount: 'Received:',
          remainingAmount: 'Remaining:',
          transactions: '{txAmount} transactions'
        }}
      />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should set txStatus to "completed" when the remaining balance is <= 0', () => {
    const wrapper = shallow(<PaymentRequest
      symbol='BTC'
      decimalPlaces={8}
      sellerName='Such Company LTD'
      logos={{
        coin: "https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg",
        company: companyAsset
      }}
      address="1BitcoinEaterAddressDontSendf59kuE"
      amount={{ toPay: 0.9, received: 0.9 }}
      transactions={[
        {
          txHash: '369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1',
          txUrl: 'https://www.blockchain.com/btc/tx/369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1',
          amount: 0.1
        },
        {
          txHash: '123',
          txUrl: 'https://www.blockchain.com/btc/tx/519f6c9581ce27e0a59f5f8e427b672087e1f2eb1aead0d66288de62ed3e9647',
          amount: 0.1
        }
      ]}
      deadline={{
        dateLocale: 'en-US',
        datetime: new Date(),
        humanized: true
      }}
      strings={{
        help: 'FooBar'
      }}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render spining animation when status is "pending" and waitAnimation is true.', () => {
    const wrapper = shallow(<PaymentRequest
      waitAnimation
      status={PaymentStatus.PENDING}
      symbol='BTC'
      decimalPlaces={8}
      logos={{
        coin: "https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
      }}
      address="1BitcoinEaterAddressDontSendf59kuE"
      amount={{ toPay: 0.9 }}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should be fully strings customizable', () => {
    const wrapper = shallow(<PaymentRequest
      symbol='BTC'
      decimalPlaces={8}
      sellerName='Such Company LTD'
      logos={{
        coin: "https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg",
        company: companyAsset
      }}
      address="1BitcoinEaterAddressDontSendf59kuE"
      amount={{ toPay: 0.9, received: 0.2 }}
      transactions={[
        {
          txHash: '369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1',
          txUrl: 'https://www.blockchain.com/btc/tx/369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1',
          amount: 0.1
        },
        {
          txHash: '519f6c9581ce27e0a59f5f8e427b672087e1f2eb1aead0d66288de62ed3e9647',
          txUrl: 'https://www.blockchain.com/btc/tx/519f6c9581ce27e0a59f5f8e427b672087e1f2eb1aead0d66288de62ed3e9647',
          amount: 0.1
        }
      ]}
      helpUrl='https://github.com/PaulFasola/adoption/blob/master/README.md'
      onCancel={() => alert('User wants to cancel, do something here!')}
      deadline={{
        dateLocale: 'en-US',
        datetime: new Date(2009, 1, 3)
      }}
      customStatusText={'Custom status message'}
      strings={{
        cancel: 'Very cancelling',
        seller: 'Seller',
        deadline: 'Deadline',
        // txStatus is overriden by customStatusText props in this example.
        txStatus: {
          complete: 'Done \o/',
          failed: 'Failed :(',
          pending: 'is pending...'
        },
        request: 'Pay {amount} {symbol} to get your item',
        help: 'Wow, help here',
        status: 'Transaction',
        receivedAmount: 'Received:',
        remainingAmount: 'Remaining:',
        transactions: '{txAmount} transactions'
      }} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

const PaymentRequestComponent = (<PaymentRequest
  symbol='BTC'
  decimalPlaces={8}
  logos={{
    coin: "https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
  }}
  transactions={
    // Spawn 30 txs
    new Array<ITransaction>(30).fill({
      txHash: '369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1',
      txUrl: 'https://www.blockchain.com/btc/tx/369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1',
      amount: 0.1
    }, 0, 20)
  }
  address="1BitcoinEaterAddressDontSendf59kuE"
  strings={{
    help: 'Halp!'
  }}
  amount={{ toPay: 0.9 }} />);

describe(`PaymentRequest - Edge cases`, () => {
  it('should prevent transaction container to overflow', () => {
    const component = render(PaymentRequestComponent);
    const lastLi = component.container.querySelector('li:last-child') as HTMLLIElement;
    if (!lastLi) fail();
    expect(lastLi.offsetParent).toBe(null); // null offsetParent = the element is not visible
  });
});