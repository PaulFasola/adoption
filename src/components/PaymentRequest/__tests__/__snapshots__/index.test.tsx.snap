// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentRequest Basic should not render any visual status if 1) payment status is not "completed" or "failed" and 2) showQRCode prop is false. 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="BTC logo"
      src="https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
    />
  </styled.div>
  <div>
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": 0.9,
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="Please send {amount} {symbol} to address:"
      />
      <div
        title="1BitcoinEaterAddressDontSendf59kuE"
      >
        1BitcoinEaterAddressDontSendf59kuE
      </div>
    </styled.div>
    <styled.div />
    <styled.div>
      <Item
        title="Current status"
      >
        <AdaptiveSpan
          showTitle={true}
          style="bold"
          text="Waiting for payment"
        />
      </Item>
    </styled.div>
  </div>
  <div />
</styled.div>
`;

exports[`PaymentRequest Basic should render a visual animation if status is "completed" or "failed" 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="BTC logo"
      src="https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
    />
  </styled.div>
  <div>
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": 0.9,
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="Please send {amount} {symbol} to address:"
      />
      <div
        title="1BitcoinEaterAddressDontSendf59kuE"
      >
        1BitcoinEaterAddressDontSendf59kuE
      </div>
    </styled.div>
    <styled.div>
      <AnimatedIcon
        key="complete"
        style={
          Object {
            "display": "flex",
            "margin": "auto",
          }
        }
        type="success"
      />
    </styled.div>
    <styled.div>
      <Item
        title="Current status"
      >
        <AdaptiveSpan
          showTitle={true}
          style="bold"
          text="Payment complete"
        />
      </Item>
    </styled.div>
  </div>
  <div />
</styled.div>
`;

exports[`PaymentRequest Basic should render a visual animation if status is "completed" or "failed" 2`] = `
<styled.div>
  <styled.div>
    <img
      alt="BTC logo"
      src="https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
    />
  </styled.div>
  <div>
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": 0.9,
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="Please send {amount} {symbol} to address:"
      />
      <div
        title="1BitcoinEaterAddressDontSendf59kuE"
      >
        1BitcoinEaterAddressDontSendf59kuE
      </div>
    </styled.div>
    <styled.div>
      <AnimatedIcon
        key="failed"
        style={
          Object {
            "display": "flex",
            "margin": "auto",
          }
        }
        type="failure"
      />
    </styled.div>
    <styled.div>
      <Item
        title="Current status"
      >
        <AdaptiveSpan
          showTitle={true}
          style="bold"
          text="Payment failed"
        />
      </Item>
    </styled.div>
  </div>
  <div />
</styled.div>
`;

exports[`PaymentRequest Basic should render relative time correctly (days) 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="BTC logo"
      src="https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
    />
  </styled.div>
  <div>
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": 0.9,
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="Please send {amount} {symbol} to address:"
      />
      <div
        title="1BitcoinEaterAddressDontSendf59kuE"
      >
        1BitcoinEaterAddressDontSendf59kuE
      </div>
    </styled.div>
    <styled.div>
      <Styled(QRCode)
        bgColor="transparent"
        fgColor="#000000"
        level="L"
        style={
          Object {
            "width": 100,
          }
        }
        value="1BitcoinEaterAddressDontSendf59kuE"
      />
    </styled.div>
    <styled.div>
      <Item
        title="Send before"
      >
        2/5/2009
        <span>
           (
          in 2 days
          )
        </span>
      </Item>
      <Item
        title="Current status"
      >
        <AdaptiveSpan
          showTitle={true}
          style="bold"
          text="Waiting for payment"
        />
      </Item>
    </styled.div>
  </div>
  <div />
</styled.div>
`;

exports[`PaymentRequest Basic should render relative time correctly (hours) 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="BTC logo"
      src="https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
    />
  </styled.div>
  <div>
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": 0.9,
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="Please send {amount} {symbol} to address:"
      />
      <div
        title="1BitcoinEaterAddressDontSendf59kuE"
      >
        1BitcoinEaterAddressDontSendf59kuE
      </div>
    </styled.div>
    <styled.div>
      <Styled(QRCode)
        bgColor="transparent"
        fgColor="#000000"
        level="L"
        style={
          Object {
            "width": 100,
          }
        }
        value="1BitcoinEaterAddressDontSendf59kuE"
      />
    </styled.div>
    <styled.div>
      <Item
        title="Send before"
      >
        2/3/2009
        <span>
           (
          in 2 hours
          )
        </span>
      </Item>
      <Item
        title="Current status"
      >
        <AdaptiveSpan
          showTitle={true}
          style="bold"
          text="Waiting for payment"
        />
      </Item>
    </styled.div>
  </div>
  <div />
</styled.div>
`;

exports[`PaymentRequest Basic should render relative time correctly (minutes) 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="BTC logo"
      src="https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
    />
  </styled.div>
  <div>
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": 0.9,
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="Please send {amount} {symbol} to address:"
      />
      <div
        title="1BitcoinEaterAddressDontSendf59kuE"
      >
        1BitcoinEaterAddressDontSendf59kuE
      </div>
    </styled.div>
    <styled.div>
      <Styled(QRCode)
        bgColor="transparent"
        fgColor="#000000"
        level="L"
        style={
          Object {
            "width": 100,
          }
        }
        value="1BitcoinEaterAddressDontSendf59kuE"
      />
    </styled.div>
    <styled.div>
      <Item
        title="Send before"
      >
        2/3/2009
        <span>
           (
          in 2 minutes
          )
        </span>
      </Item>
      <Item
        title="Current status"
      >
        <AdaptiveSpan
          showTitle={true}
          style="bold"
          text="Waiting for payment"
        />
      </Item>
    </styled.div>
  </div>
  <div />
</styled.div>
`;

exports[`PaymentRequest Detailed Should be fully strings customizable 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="BTC logo"
      src="https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
    />
    <img
      alt="Company logo"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </styled.div>
  <div>
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": 0.9,
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="Please send {amount} {symbol} to address:"
      />
      <div
        title="1BitcoinEaterAddressDontSendf59kuE"
      >
        1BitcoinEaterAddressDontSendf59kuE
      </div>
    </styled.div>
    <styled.div>
      <Styled(QRCode)
        bgColor="transparent"
        fgColor="#000000"
        level="L"
        style={
          Object {
            "width": 100,
          }
        }
        value="1BitcoinEaterAddressDontSendf59kuE"
      />
    </styled.div>
    <styled.div>
      <Item
        title="Merchant"
      >
        <span
          title="Such Company LTD"
        >
          Such Company LTD
        </span>
      </Item>
      <Item
        title="Amount received"
      >
        0.200000
         
        BTC
      </Item>
      <Item
        title="Amount remaining"
      >
        0.700000
         
        BTC
      </Item>
      <Item
        title="Send before"
      >
        2/3/2009
      </Item>
      <Item
        title="Transactions (2)"
      >
        <styled.ul>
          <li
            key="0"
          >
            <a
              href="https://www.blockchain.com/btc/tx/369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1"
              rel="noreferrer"
              target="_blank"
              title="Transaction"
            >
              369...cf1
            </a>
            <i>
              (
              0.100000
               
              BTC
              )
            </i>
          </li>
          <li
            key="1"
          >
            <a
              href="https://www.blockchain.com/btc/tx/519f6c9581ce27e0a59f5f8e427b672087e1f2eb1aead0d66288de62ed3e9647"
              rel="noreferrer"
              target="_blank"
              title="Transaction"
            >
              519...647
            </a>
            <i>
              (
              0.100000
               
              BTC
              )
            </i>
          </li>
        </styled.ul>
      </Item>
      <Item
        title="Current status"
      >
        <AdaptiveSpan
          showTitle={true}
          style="bold"
          text="Custom status message"
        />
      </Item>
    </styled.div>
  </div>
  <div>
    <styled.button
      onClick={[Function]}
    >
      Cancel
    </styled.button>
    <styled.a
      href="https://github.com/PaulFasola/adoption/blob/master/README.md"
      target="blank"
      title="Need help? Click here!"
    >
      <Icon
        style={
          Object {
            "lineHeight": "15px",
            "width": 13,
          }
        }
        type="help-circle-o"
      />
      Need help? Click here!
    </styled.a>
  </div>
</styled.div>
`;

exports[`PaymentRequest Detailed should be fully string customizable 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="BTC logo"
      src="https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
    />
  </styled.div>
  <div>
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": 0.9,
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="Please send {amount} {symbol} to address:"
      />
      <div
        title="1BitcoinEaterAddressDontSendf59kuE"
      >
        1BitcoinEaterAddressDontSendf59kuE
      </div>
    </styled.div>
    <styled.div>
      <Styled(QRCode)
        bgColor="transparent"
        fgColor="#000000"
        level="L"
        style={
          Object {
            "width": 100,
          }
        }
        value="1BitcoinEaterAddressDontSendf59kuE"
      />
    </styled.div>
    <styled.div>
      <Item
        title="Current status"
      >
        <AdaptiveSpan
          showTitle={true}
          style="bold"
          text="Waiting for payment"
        />
      </Item>
    </styled.div>
  </div>
  <div />
</styled.div>
`;

exports[`PaymentRequest Detailed should render spining animation when status is "pending" and waitAnimation is true. 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="BTC logo"
      src="https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
    />
  </styled.div>
  <div>
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": 0.9,
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="Please send {amount} {symbol} to address:"
      />
      <div
        title="1BitcoinEaterAddressDontSendf59kuE"
      >
        1BitcoinEaterAddressDontSendf59kuE
      </div>
    </styled.div>
    <styled.div>
      <Styled(QRCode)
        bgColor="transparent"
        fgColor="#000000"
        level="L"
        style={
          Object {
            "width": 100,
          }
        }
        value="1BitcoinEaterAddressDontSendf59kuE"
      />
    </styled.div>
    <styled.div>
      <Item
        title="Current status"
      >
        <styled.div />
        <AdaptiveSpan
          showTitle={true}
          style="bold"
          text="Waiting for payment"
        />
      </Item>
    </styled.div>
  </div>
  <div />
</styled.div>
`;

exports[`PaymentRequest Detailed should set txStatus to "completed" when the remaining balance is <= 0 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="BTC logo"
      src="https://upload.wikimedia.org/wikipedia/commons/c/c5/Bitcoin_logo.svg"
    />
    <img
      alt="Company logo"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </styled.div>
  <div>
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": 0.9,
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="Please send {amount} {symbol} to address:"
      />
      <div
        title="1BitcoinEaterAddressDontSendf59kuE"
      >
        1BitcoinEaterAddressDontSendf59kuE
      </div>
    </styled.div>
    <styled.div>
      <Styled(QRCode)
        bgColor="transparent"
        fgColor="#000000"
        level="L"
        style={
          Object {
            "width": 100,
          }
        }
        value="1BitcoinEaterAddressDontSendf59kuE"
      />
    </styled.div>
    <styled.div>
      <Item
        title="Merchant"
      >
        <span
          title="Such Company LTD"
        >
          Such Company LTD
        </span>
      </Item>
      <Item
        title="Amount received"
      >
        0.900000
         
        BTC
      </Item>
      <Item
        title="Amount remaining"
      >
        0.000000
         
        BTC
      </Item>
      <Item
        title="Transactions (2)"
      >
        <styled.ul>
          <li
            key="0"
          >
            <a
              href="https://www.blockchain.com/btc/tx/369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1"
              rel="noreferrer"
              target="_blank"
              title="Transaction"
            >
              369...cf1
            </a>
            <i>
              (
              0.100000
               
              BTC
              )
            </i>
          </li>
          <li
            key="1"
          >
            <a
              href="https://www.blockchain.com/btc/tx/519f6c9581ce27e0a59f5f8e427b672087e1f2eb1aead0d66288de62ed3e9647"
              rel="noreferrer"
              target="_blank"
              title="Transaction"
            >
              123
            </a>
            <i>
              (
              0.100000
               
              BTC
              )
            </i>
          </li>
        </styled.ul>
      </Item>
      <Item
        title="Current status"
      >
        <AdaptiveSpan
          showTitle={true}
          style="bold"
          text="Payment complete"
        />
      </Item>
    </styled.div>
  </div>
  <div />
</styled.div>
`;
